pkg/deploy/kapp.go

```diff
@@ -162,6 +163,12 @@ func (a *Kapp) addDeployArgs(args []string) ([]string, error) {
 		args = append(args, []string{"--map-ns", val}...)
 	}
 
+	// Global raw options are applied first to be able to override them within an App
+	args, err := a.addRawOpts(args, a.globalDeployRawOpts, kappAllowedDeployFlagSet)
```

Comment:
it feels funny to call the same fn twice in a row,  especially because the first time we do an error check but the second time we can just return whatever it returned.  Is it worth it to add a handler like, :

```
a.multiAddRawOpts(args, []RawOptions, FlagSet) args, error {
  var args;
   for opts in RawOptions {
       args, err := a.addRawOpts(args, opts, FlagSet)
       return err if err != nil
     }
     return args, nil
}
```

just to hide the sort of double-call? 

(forgive my pseudo code above. hopefully gets the gist across)

Reply:
ill disagree with this one. less indirection reads clearer.